### PR TITLE
Cxf-soap tests: Refactor and split

### LINKED ISSUE #4171: Cxf-soap tests: Refactor and split
It would be nice to:

1. Split tests for `cxf-soap` into modules for `client`, `server` and `ws-security`.
2. Migrate the Client test from Wiremock to a testcontainer like it is done in https://github.com/quarkiverse/quarkus-cxf/tree/main/integration-tests/client

### PR DESCRIPTION
**Pull request by ppalaga**
Fix #4171

Replaces #4172 and #4187 by @JiriOndrusek (adding the suggestions mentioned in https://github.com/apache/camel-quarkus/pull/4187#issuecomment-1282564510 )

### CODE REVIEW

**Review by oscerd — APPROVED** [review]

File: integration-test-groups/cxf-soap/cxf-soap-client/src/main/resources/wsdl/CalculatorService.wsdl [hunk 1]
```diff
@@ -0,0 +1,248 @@
+<?xml version="1.0" ?><wsdl:definitions xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:wsdl="http://schemas.xmlsoap.org/wsdl/" xmlns:tns="http://www.jboss.org/eap/quickstarts/wscalculator/Calculator" xmlns:soap="http://schemas.xmlsoap.org/wsdl/soap/" xmlns:ns1="http://schemas.xmlsoap.org/soap/http" name="CalculatorService" targetNamespace="http://www.jboss.org/eap/quickstarts/wscalculator/Calculator">
+<!--
```
Comment (zbendhiba) [inline integration-test-groups/cxf-soap/cxf-soap-client/src/main/resources/wsdl/CalculatorService.wsdl:2]: Is there a reason why this is not the first line of the file ?

Comment (zbendhiba) [reply integration-test-groups/cxf-soap/cxf-soap-client/src/main/resources/wsdl/CalculatorService.wsdl:2]: it's the same in XML files, ignore my comment

**Review by zbendhiba — APPROVED** [review]
LGTM

**Comment by jamesnetherton** [comment]
@ppalaga I'm guessing this needs backporting to 2.13.x?

**Pull request merged**